apps/server/src/server/repositories/cache/redis-cache.repository.ts

```diff
@@ -101,6 +102,40 @@ export class RedisCacheRepository implements CacheRepository {
     return this.getArray<Tx>('transactions')
   }
 
+  public putExchangeRate(exchangerate: ExchangeRate): Promise<boolean> {
+    return new Promise(resolve => {
+      this.redis
+        .set(exchangerate.base, exchangerate)
+        .then(result => {
+          if (!result) {
+            resolve(false)
+          }
+          resolve(true)
+        })
+        .catch(error => {
+          resolve(false)
+        })
+    })
+  }
+
+  public getExchangeRate(token: string): Promise<ExchangeRate> {
+    return new Promise((resolve, reject) => {
+      this.redis
+        .get(token)
+        .then(result => {
+          if (!result) {
+            reject()
+          } else {
```

Comment:
Rewrite to:

if (!result) {
reject()
return
}
const val = JSON.parse(result) as ExchangeRate
resolve(val)

Try to avoid else clauses at any cost (much better to understand code when is less indented)